src/server/snapshot.cc

```diff
@@ -268,6 +276,7 @@ unsigned SliceSnapshot::SerializeBucket(DbIndex db_index, PrimeTable::bucket_ite
 
   while (!it.is_done()) {
     ++result;
+    // might yield
```

Comment:
could you check this changes with a temporary change here (not to be pushed) that calls yield here?